ert_shared/ensemble_evaluator/queue_adaptor.py

```diff
@@ -0,0 +1,121 @@
+import asyncio
+import json
+import threading
+import time
+from concurrent.futures import thread
+from pathlib import Path
+
+import websockets
+from ert_shared.ensemble_evaluator.entity import create_realization
+from ert_shared.ensemble_evaluator.ws_util import wait_for_ws
+from res.job_queue import JobQueueManager
+from res.job_queue.job_status_type_enum import JobStatusType
+
+
+def status_equal(statuses):
+    return statuses[0] == statuses[1]
+
+
+class _EventDecoder(json.JSONEncoder):
+    def default(self, o):
+        return o.__dict__
+
+
+class JobQueueManagerAdaptor(JobQueueManager):
+
+    ws_url = "ws://localhost:8765"
+
+    def __init__(self, queue, queue_evaluators=None):
+        super().__init__(queue, queue_evaluators)
+        self._ws_url = self.ws_url
+        self._changes_queue = asyncio.Queue()
+
+        wait_for_ws(self._ws_url)
+        self._initialize_state()
+
+        # JobQueueManager is at its core a while and a sleep. This is not
+        # asyncio friendly, so create a separate thread from where websocket
+        # events will be pushed.
+        self._publisher_thread = threading.Thread(
+            target=self._publisher, args=(asyncio.get_event_loop(),)
+        )
+        self._publisher_thread.start()
+
+    def _publisher(self, loop):
+        loop.run_until_complete(
+            self._publish_from_queue(self._changes_queue, self._ws_url)
+        )
+
+    @staticmethod
+    async def _publish_from_queue(queue, url):
+        while True:
+            async with websockets.connect(url) as websocket:
+                changes = await queue.get()
+                event = json.dumps(changes, cls=_EventDecoder)
+                await websocket.send(event)
+                if changes is None:
+                    return
+
+    def _initialize_state(self):
+        self._qindex_to_iens = {
+            q_index: q_node.callback_arguments[0].iens
+            for q_index, q_node in enumerate(self._queue.job_list)
+        }
```

Comment:
Think it is good practice to set all member variables in the init method, so that its easy to see all of them.  (Could for example set it to None in init and then have everything else as is)